Rules/CommonScripts/RegrowPlants.as

```diff
@@ -292,7 +292,15 @@ void onTick(CRules@ this)
 
 				if (random_grow <= chicken_grow_chance && chicken_count < chicken_limit)
 				{
-					server_CreateBlob("chicken", -1, tinfo.coords - Vec2f(0,tilesize));
+					Vec2f spawn_position = tinfo.coords - Vec2f(0,tilesize);
+				
+					if (map.isBlobInRadius("mine", spawn_position, 16.0f) // don't spawn chicken inside mines
+						|| map.isInWater(spawn_position)) // don't spawn chicken inside water	
+					{
+						break;
```

Comment:
This is going to break out from the entire tile loop, and we don't want that.

Reply:
It should be better now.